src/engine/runtime.js

```diff
@@ -1232,63 +1256,102 @@ class Runtime extends EventEmitter {
             argTypeInfo = context.categoryInfo.customFieldTypes[argInfo.type].argumentTypeInfo;
         }
 
-        const defaultValue =
-            typeof argInfo.defaultValue === 'undefined' ? '' :
-                xmlEscape(maybeFormatMessage(argInfo.defaultValue, this.makeMessageContextForTarget()).toString());
+        // Start to construct the scratch-blocks style JSON defining how the block should be
+        // laid out
+        let argJSON;
 
-        if (argTypeInfo.check) {
-            argJSON.check = argTypeInfo.check;
+        // Most field types are inputs (slots on the block that can have other blocks plugged into them)
+        // check if this is not one of those cases. E.g. an inline image on a block.
+        switch (argTypeInfo.fieldType) {
+        case 'field_image': {
+            if (!argInfo.dataURI) {
+                log.warn('Missing data URI in extension block with argument type IMAGE');
+            }
+            argJSON = {
+                type: 'field_image',
+                src: argInfo.dataURI || '',
+                // TODO these probably shouldn't be hardcoded...?
+                width: 24,
+                height: 24,
```

Comment:
Hmm... I think it makes sense to specify the height so that it fits the block correctly. If you leave out the width will it auto-scale appropriately, maintaining the aspect ratio?

Reply:
Oh mostly I had an issue with the numbers being hardcoded here instead of constants we pull from somewhere (esp. if we use the VM with a version of scratch-blocks that has some other settings). The number `24` has no context here.